indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/symmetric/SymmetricKey.kt

```diff
@@ -0,0 +1,211 @@
+            ) : NonAuthenticating<NonceTrait.Without>(algorithm, secretKey)
+        }
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (other !is Integrated<*, *>) return false
+
+            if (algorithm != other.algorithm) return false
+            if (!secretKey.contentEquals(other.secretKey)) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = algorithm.hashCode()
+            result = 31 * result + secretKey.contentHashCode()
+            return result
+        }
+    }
+
+
+    /**
+     * Encryption key with dedicated MAC key. Used for non-authenticated ciphers that use an external MAC function to
+     * bolt on AEAD capabilities, such as [SymmetricEncryptionAlgorithm.AES.GCM]
+     */
+    sealed class WithDedicatedMac<I : NonceTrait>
+    @HazardousMaterials("Does not check whether key size matched algorithm! Useful for testing, but not production!")
+    /**
+     * Do not invoke directly! use Supreme's `SymmetricEncryptionAlgorithm.randomKey()` and `SymmetricEncryptionAlgorithm.encryptionKeyFrom(bytes)`
+     * This constructor does not check for matching key sizes to allow for testing error cases!
+     */
+    constructor(
+        override val algorithm: SymmetricEncryptionAlgorithm<AuthCapability.Authenticated.WithDedicatedMac<*, I>, I, KeyType.WithDedicatedMacKey>,
+        /**
+         * The actual encryption key bytes
+         */
+        val encryptionKey: ByteArray,
+        /**
+         * The actual dedicated MAX key bytes
+         */
+        val macKey: ByteArray
+    ) : SymmetricKey<AuthCapability.Authenticated.WithDedicatedMac<*, I>, I, KeyType.WithDedicatedMacKey>,
+        SymmetricKey.Authenticating<AuthCapability.Authenticated.WithDedicatedMac<*, I>, I, KeyType.WithDedicatedMacKey> {
+        class RequiringNonce(
+            algorithm: SymmetricEncryptionAlgorithm<AuthCapability.Authenticated.WithDedicatedMac<*, NonceTrait.Required>, NonceTrait.Required, KeyType.WithDedicatedMacKey>,
+            secretKey: ByteArray,
+            dedicatedMacKey: ByteArray,
+            override val additionalProperties: MutableMap<String, String> = mutableMapOf<String, String>()
+        ) : WithDedicatedMac<NonceTrait.Required>(
+            algorithm, secretKey, dedicatedMacKey
+        ),
+            SymmetricKey.RequiringNonce<AuthCapability.Authenticated.WithDedicatedMac<*, NonceTrait.Required>, KeyType.WithDedicatedMacKey>
+
+        class WithoutNonce(
+            algorithm: SymmetricEncryptionAlgorithm<AuthCapability.Authenticated.WithDedicatedMac<*, NonceTrait.Without>, NonceTrait.Without, KeyType.WithDedicatedMacKey>,
+            secretKey: ByteArray,
+            dedicatedMacKey: ByteArray,
+            override val additionalProperties: MutableMap<String, String> = mutableMapOf<String, String>()
```

Comment:
same as above, why is this declared on every subclass instead of on the superclass

Reply:
fixed